src/System.Json/src/System/Json/JsonValue.cs

```diff
@@ -70,26 +71,14 @@ private static JsonValue ToJsonValue(object ret)
             }
 
             if (ret is bool) return new JsonPrimitive((bool)ret);
-            if (ret is byte) return new JsonPrimitive((byte)ret);
```

Comment:
Sure -
Basically, System.Json uses an internal `JavaScript` reader to parse data. Firstly, it is internal so not extensible.
The parsing functionality is found here: https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L39

Basically, the method can only return the following:
- a List<object>: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L57)
- object[]: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L76)
- Dictionary<string, object>: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L85)
- object[]: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L112)
- bool: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L116) or [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L120)
- string: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L127)
- int: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L260)
- ulong: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L272)
- decimal: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L279)
- double: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L318)